pkg/agent/protocols/s3.go

```diff
@@ -0,0 +1,99 @@
+package protocols
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type S3Manager struct {
+	S3 *s3.S3
+}
+
+func (m *S3Manager) Download(modelDir string, storageUri string) error {
+	s3Uri := strings.TrimPrefix(storageUri, string(S3))
+	path := strings.SplitN(s3Uri, "/", 2)
+	bucket := path[0]
+	item := path[1]
+	s3ObjectLoactor := &S3ObjectDownloader{
```

Comment:
nit: loactor

Reply:
Why are you giving two names to this thing? locator and downloader. Which is it?